lib/kintsugi/cli.rb

```diff
@@ -88,9 +77,23 @@ def create_install_driver_subcommand
         end
 
         driver_options = options.map do |option_name, option_value|
-          prefix = option_name.length > 1 ? "--" : "-"
-          [prefix + option_name.to_s, option_value]
-        end.flatten
+          switch = option_parser.top.search(:long, option_name.to_s)
+          prefix = "--"
+          if switch.nil?
+            switch = option_parser.top.search(:short, option_name.to_s)
+            prefix = "-"
+          end
+          case switch
+          when OptionParser::Switch::NoArgument
+            [prefix + option_name.to_s]
+          when NilClass
+            puts "Invalid flag #{option_name} passed to 'install-driver' subcommand\n\n"
+            puts option_parser
+            exit(1)
+          else
+            [prefix + option_name.to_s, option_value]
+          end
+        end.compact.flatten
```

Comment:
Is `compact` necessary here?  It should never happen with the current code

Reply:
correct